src/lib.rs

```diff
@@ -197,7 +197,7 @@ impl Progress {
                 let mut b = &mut self.bars[bar.0];
                 let w = (term_width / 2) - 7;
                 let (data, unit) = denomination(b.curr);
-                let diff = 100 * (b.curr - b.prev) / b.total;
+                let diff = 100 * ((b.curr - b.prev) as u64 / (b.total as u64));
```

Comment:
the issue is the parentheses. `100 * (x / total)` can only result in 0 or 100, nothing in between when `x` and `total` are integers and `x` <= `total`.

Reply:
I think this should work:

```suggestion
                let diff = (100 * (b.curr - b.prev) as u64) / (b.total as u64);
```